index.ts

```diff
@@ -0,0 +1,25 @@
+import type { InputManifest, Result } from '@balena/transformer-sdk'
+import { v4 as uuidv4 } from 'uuid';
+import type { ExternalEventContract, DiagnosticEventContractDefinition } from './types'
+
+export async function externalevent2diagnosticevent (input: InputManifest<ExternalEventContract>): Promise<Result<DiagnosticEventContractDefinition>[]> {
+    const diagnosticEvent: DiagnosticEventContractDefinition = {
+        type: 'diagnostic-event',
+        slug: `diagnostic-event-${uuidv4()}`,
+        data: {
+            fingerprint: input.input.contract.data.payload.fingerprint,
+            priority: input.input.contract.data.payload.priority,
+            message: input.input.contract.data.payload.message,
+            timestamp: input.input.contract.data.payload.timestamp,
+        },
+        artifact: {
+            '$export': 'foobar',
+        },
```

Comment:
@codewithcheese What should this be?

Reply:
@codewithcheese Getting a type error, claiming that it is required:
![type-error](https://user-images.githubusercontent.com/45343541/166418540-71c3828d-59aa-46d8-aa21-4253295d9ade.png)

Reply:
@joshbwlng ahh my bad, https://github.com/product-os/transformer-sdk/pull/12